integrationtests/src/test/java/io/kroxylicious/proxy/filter/CreateTopicFilter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import org.apache.kafka.common.message.CreateTopicsRequestData;
+import org.apache.kafka.common.message.CreateTopicsResponseData;
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.kroxylicious.proxy.config.BaseConfig;
+
+public class CreateTopicFilter implements CreateTopicsRequestFilter, CreateTopicsResponseFilter {
+
+    private final String name;
+
+    public CreateTopicFilter(CreateTopicFilterConfig config) {
+        name = config.name;
+    }
+
+    public static class CreateTopicFilterConfig extends BaseConfig {
+        private final String name;
+
+        @JsonCreator
+        public CreateTopicFilterConfig(@JsonProperty("name") String name) {
+            this.name = name;
+        }
+    }
+
+    @Override
+    public void onCreateTopicsRequest(short apiVersion, RequestHeaderData header, CreateTopicsRequestData request, KrpcFilterContext context) {
+        header.setClientId(header.clientId() + "-" + name);
+        context.forwardRequest(header, request);
+    }
+
+    @Override
+    public void onCreateTopicsResponse(short apiVersion, ResponseHeaderData header, CreateTopicsResponseData response, KrpcFilterContext context) {
+        CreateTopicsResponseData.CreatableTopicResult newElement = new CreateTopicsResponseData.CreatableTopicResult();
+        newElement.setName(name);
```

Comment:
I assume this is just marking the response so that you can detect it has been through the filter.  Wonder if untagged unknownTaggedFields might be offer a less intrusive way to achieve the same ends.   You could also take the same approach on the request/response.

Reply:
Rather than `CreateTopicFilter`, how about a `RequestResponseMarkingFilter`?
It would just add an `unknownTaggedField` (a mark) to request/response as the request/response passes through as it goes through.

* RequestResponseMarkingFilter would implement RequestFilter and ResponseFilter
* It would take a set of ApiKeys - the set of keys to intercept.
* It would call `org.apache.kafka.common.protocol.Message#unknownTaggedFields()` to mark the request/response as it goes though with the name.

I think this filter might have better reuse potential.

Reply:
Done, yep I think it's clearer this way.